extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php

```diff
@@ -34,11 +48,68 @@ static public function onImageServingCollectImages( &$imageNamesArray, $articleT
 	 * @param $parser
 	 * @param $frame
 	 */
-	static public function onParserTagHooksBeforeInvoke( $name, $marker, $content, $attributes, $parser, $frame ) {
-		if ( $name === self::PARSER_TAG_GALLERY ) {
-			\Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()->setGallery( $marker, $content );
+	static public function onParserTagHooksBeforeInvoke($name, $marker, $content, $attributes, $parser, $frame) {
+		if ($name === self::PARSER_TAG_GALLERY) {
+			\Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()->setGallery($marker, $content);
+		}
+
+		return true;
+	}
+
+	/**
+	 * renders portable infobox builder instead of edit page while creating new template
+	 *
+	 * @param EditPageLayoutController $editPageContext
+	 * @return bool
+	 */
+	static public function onEditPageLayoutExecute( $editPageContext ) {
+		global $wgRequest;
+
+		if ( self::isEditingNewTemplate() && $wgRequest->getVal( 'portableInfoboxBuilder', false ) ) {
```

Comment:
Would it be nice to extract this param name to some const? AFAIK it can be changed soon so it will be easier to do it if we'll keep it in one place.